src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -95,28 +95,25 @@ export class ESLSwipeGestureTarget extends SyntheticEventTarget {
    */
   @bind
   protected handleEnd(e: PointerEvent): void {
-    // if the user released on a different target, cancel!
-    if (!this.isGestureStarted || (this.startEvent?.target !== e.target)) return;
 
     const eventsDiff = this.eventDiff(e);
     const direction = this.resolveDirection(eventsDiff);
     if (direction) {
       const swipeInfo: ESLSwipeGestureEventInfo = {
         direction,
-        distanceX: Math.abs(eventsDiff.x),
-        distanceY: Math.abs(eventsDiff.y),
+        distanceX: eventsDiff.x,
+        distanceY: eventsDiff.y,
+        distance: Math.round(Math.hypot(eventsDiff.x, eventsDiff.y, 2)),
+        angle: Math.round((Math.atan2(eventsDiff.y, eventsDiff.x) * 180 / Math.PI + 90 + 360) % 360),
         startEvent: this.startEvent,
         endEvent: e
       };
 
       // fire `swipe` event on the element that started the swipe
       this.dispatchEvent(ESLSwipeGestureEvent.fromConfig('swipe', this.target, swipeInfo));
-      // fire `swipe:${dir}` event on the element that started the swipe
-      this.dispatchEvent(ESLSwipeGestureEvent.fromConfig(`swipe:${direction}`, this.target, swipeInfo));
     }
 
-    // Mark swipe as completed
-    this.isGestureStarted = false;
+    this.target.releasePointerCapture && this.target.releasePointerCapture(e.pointerId);
```

Comment:
We can free pointer on the start (so in case something throw error (e.g. user handlrer) the UI will not break for user)